UnitTests/UnitTests.UWP/UI/Controls/Test_Carousel.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using Windows.UI.Xaml.Automation;
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.AppContainer;
+using Windows.UI.Xaml.Automation.Peers;
+using Microsoft.Toolkit.Uwp.UI.Automation.Peers;
+
+namespace UnitTests.UWP.UI.Controls
+{
+    [TestClass]
+    [TestCategory("Test_Carousel")]
+    public class Test_Carousel
+    {
+        [UITestMethod]
+        public void ShouldConfigureCarouselAutomationPeer()
```

Comment:
Can we also test that selecting works and position in set and size of set are accurate?

Reply:
I've been trying to look into this for some days and am coming up short. I believe because the actual UI is not rendered in the UWP tests, the containers for the items are never prepared which means that the `CarouselItem` instances are never created.

I may be missing something on these though which is right in front of me and potentially others have written these types of tests for `ItemsControl` based controls before that could help.

Reply:
Do you have a way in the WCT testapp, to actually render the items? The WinUI API tests (which this kind of test would be) have a simple API that allows you to set the content of the app, thus forcing items to be rendered and having the visual tree be populated.

Reply:
@chingucoding @jamesmcroft yes, the new UI Integration Tests (using the WinUI infrastructure) are running an actual app, so we can host whatever we want there in an example page and check whatever we need to is realized from there (unlike the normal Unit Test environment which is only really running on a UI Thread but without visuals).

Reply:
Is there a specific reason why the UI thread can't render visuals? You can easily create unit tests which render UI as part of the test so I'm a bit confused why this is not an option for WCT tests yet.